plasmapy/diagnostics/thomson.py

```diff
@@ -1,37 +1,268 @@
 """
-Defines the Thomson scattering analysis module as part of
-`plasmapy.diagnostics`.
+Defines the Thomson scattering analysis module as
+part of the diagnostics package.
 """
 
 __all__ = [
     "spectral_density",
+    "spectral_density_model",
 ]
 
 import astropy.constants as const
 import astropy.units as u
+import numbers
 import numpy as np
+import re
 
+from lmfit import Model
 from typing import List, Tuple, Union
 
 from plasmapy.formulary.dielectric import permittivity_1D_Maxwellian
 from plasmapy.formulary.frequencies import plasma_frequency
-from plasmapy.formulary.speeds import thermal_speed
-from plasmapy.particles import Particle
-from plasmapy.utils.decorators import validate_quantities
+from plasmapy.formulary.speeds import thermal_speed, thermal_speed_coefficients
+from plasmapy.particles import Particle, particle_mass
+from plasmapy.utils.decorators import (
+    bind_lite_func,
+    preserve_signature,
+    validate_quantities,
+)
+
+_c = const.c.si.value  # Make sure C is in SI units
+_e = const.e.si.value
+_m_p = const.m_p.si.value
+_m_e = const.m_e.si.value
```

Comment:
```suggestion
c_si_unitless = const.c.si.value
e_si_unitless = const.e.si.value
m_p_si_unitless = const.m_p.si.value
m_e_si_unitless = const.m_e.si.value
```

Just to remain consistent across all of `plasmapy`.  Obviously reference throughout the fill will also need to be updated.

Eventually we plan to add some kind of unitless module to `plasmapy.utils` that will have all the most commonly used constants.